tests/test_applets/batchProcessing/test_batchProcessingGui.py

```diff
@@ -186,20 +188,39 @@ def test_BatchProcessingGui_clear(qtbot, batch_processing_gui, role_names, filen
         assert current_data_role_widget.list_widget.count() == 0
 
 
+@contextmanager
+def does_not_raise():
```

Comment:
Any reason not to use `pytest.raises` and `contextlib.nullcontext`?

LGTM otherwise.

Reply:
In short, `pytest.raises` doesn't work ;) and I didn't know about `nullcontext`.

afaik `PyQt5` won't backpropagate exceptions to callers. So qtbot installs its own excepthook and informs you after the test ran through if any exceptions occur (by setting the test to a failure).